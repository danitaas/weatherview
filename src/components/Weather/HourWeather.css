

.HourWeather-container {
    /*width: 200px;*/
    padding: 8px;
}

.HourWeather-item {

}