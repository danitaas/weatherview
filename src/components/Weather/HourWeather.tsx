import * as React from "react";
import "./DayWeather.css";
import { IDayWeather } from "../../logic/weather";
import * as moment from "moment";
import {
    formatCentigradeFromKelvin, formatDegToCard, formatFahrenheitFromKelvin,
    formatKMHFromMS, formatMPHFromMS
} from "../../logic/maths";

export interface IProps extends Partial<IDayWeather> {

}

function HourWeather(props: IProps) {
    if (!props.dt) {
        return <div>Invalid data</div>;
    }

    //could pre-format these in redux
    const mintempstr = props.mintemp ? formatCentigradeFromKelvin(props.mintemp) + "째C" + "/" + formatFahrenheitFromKelvin(props.mintemp) + "째F" : "Unknown";
    const maxtempstr =  props.maxtemp ? formatCentigradeFromKelvin(props.maxtemp) + "째C" + "/" + formatFahrenheitFromKelvin(props.maxtemp) + "째F" : "Unknown";
    const windspeedstr =  props.windspeed ? formatKMHFromMS(props.windspeed) + "kmh" + "/" + formatMPHFromMS(props.windspeed) + "mph" : "Unknown";
    const winddegstr =  props.windspeed ? formatDegToCard(props.windspeed) : "Unknown";
    const windstr =  windspeedstr + " " + winddegstr;

    //{JSON.stringify(i)}

    return (
        <div className="HourWeather-container">
            <div className="HourWeather-item">{moment.unix(props.dt).format("HH:mm")}</div>
            <div className="HourWeather-item">{mintempstr}</div>
            <div className="HourWeather-item">{maxtempstr}</div>
            <div className="HourWeather-item">{props.conditions}</div>
            <div className="HourWeather-item">{windstr}</div>
        </div>
    );
}

export default HourWeather;
