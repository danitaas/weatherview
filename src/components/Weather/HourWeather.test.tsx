import * as React from "react";
import HourWeather from "./HourWeather";
// todo: cant use typescript import without correcting toolchain
const ReactShallowRenderer = require("react-test-renderer/shallow");
const ReactTestRenderer = require("react-test-renderer");

it("renders shallow without crashing", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<HourWeather />);
    const result = renderer.getRenderOutput();
    expect(result.type).toBe("div");
});

it("renders snapshot", () => {
    const tree = ReactTestRenderer.create(<HourWeather />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders snapshot with props", () => {
    const tree = ReactTestRenderer.create(
        <HourWeather dt={1530100800} mintemp={298.785} maxtemp={298.571} conditions={"a"} windspeed={1} winddeg={1} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
