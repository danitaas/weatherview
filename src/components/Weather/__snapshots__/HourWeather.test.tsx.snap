// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div>
  Invalid data
</div>
`;

exports[`renders snapshot with props 1`] = `
<div
  className="HourWeather-container"
>
  <div
    className="HourWeather-item"
  >
    13:00
  </div>
  <div
    className="HourWeather-item"
  >
    25.64째C/78.14째F
  </div>
  <div
    className="HourWeather-item"
  >
    25.42째C/77.76째F
  </div>
  <div
    className="HourWeather-item"
  >
    a
  </div>
  <div
    className="HourWeather-item"
  >
    3.60kmh/2.24mph N
  </div>
</div>
`;
